src/state_mgmt/state/compound_state.gd

```diff
@@ -713,10 +716,12 @@ func _goto(path: StringName, args: Dictionary = {}) -> void:
 		var state: FrayState = active_states[i]
 		if state == common_active_ancestor:
 			break
+		
+		if state is FrayCompoundState:
+			state._current_state = ""
```

Comment:
I'm sure we need to check if the check is persistent before setting the current state to empty.
```suggestion
		if state is FrayCompoundState and state.is_persistent:
			state._current_state = ""
```

Reply:
Good catch. Also, seeing this again made me realize I can just move this logic into compound_state.gd's exit method. That feels better than checking the state's type.